src/components/network/network-table.js

```diff
@@ -62,21 +54,851 @@ const useStyles = makeStyles((theme) => ({
+                {
+                    width: 400,
+                    id: 'ID',
+                    dataKey: 'id',
+                },
+                {
+                    width: 200,
+                    id: 'Name',
+                    dataKey: 'name',
+                },
+                {
+                    width: 400,
+                    id: 'VoltageLevelId',
+                    dataKey: 'voltageLevelId',
+                },
+                {
+                    width: 400,
+                    id: 'HvdcLineId',
+                    dataKey: 'hvdcLineId',
+                },
+                {
+                    width: 200,
+                    id: 'ActivePower',
+                    dataKey: 'p',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+                {
+                    width: 200,
+                    id: 'ReactivePower',
+                    dataKey: 'q',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+                {
+                    width: 200,
+                    id: 'LossFactor',
+                    dataKey: 'lossFactor',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+            ],
+        },
+
+        DANGLING_LINES: {
+            index: 13,
+            name: 'DanglingLines',
+            columns: [
+                {
+                    width: 400,
+                    id: 'ID',
+                    dataKey: 'id',
+                },
+                {
+                    width: 200,
+                    id: 'Name',
+                    dataKey: 'name',
+                },
+                {
+                    width: 400,
+                    id: 'VoltageLevelId',
+                    dataKey: 'voltageLevelId',
+                },
+                {
+                    width: 200,
+                    id: 'UcteXnodeCode',
+                    dataKey: 'ucteXnodeCode',
+                },
+                {
+                    width: 200,
+                    id: 'ActivePower',
+                    dataKey: 'p',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+                {
+                    width: 200,
+                    id: 'ReactivePower',
+                    dataKey: 'q',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+                {
+                    width: 200,
+                    id: 'ConstantActivePower',
+                    dataKey: 'p0',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+                {
+                    width: 200,
+                    id: 'ConstantReactivePower',
+                    dataKey: 'q0',
+                    numeric: true,
+                    fractionDigits: 1,
+                },
+            ],
+        },
+    };
+
+    let cols = [];
```

Comment:
move this code outside of the render function, or use the useState function that takes a lambda as parameter to compute it only once